tests/manifold-map.cc

```diff
@@ -187,27 +245,27 @@ BOOST_AUTO_TEST_CASE_TEMPLATE (manifold_map_test_0, T, functionTypes_t)
       input(i) = 1 + i;
     }
 
-  (*output) << instWrap << "\n";
-  std::cout << instWrap << std::endl;
-  std::cout << "input: " << input.transpose() << std::endl;
+  (*output) << instWrap << roboptim::iendl;
   instWrap(result, input);
-  std::cout << "result: " << result.transpose() << std::endl << std::endl;
 
   for (int i = 0; i < 10; ++i)
     {
-      std::cout << "i: " << i << std::endl;
       instWrap.gradient(gradient, input, i);
     }
 
   instWrap.jacobian(jacobian, input);
-  std::cout << "jacobian: " << std::endl << jacobian << std::endl;
-  instWrap.manifold_jacobian(refJacobian, input);
-
-  (*output) << descWrapPtr;
-
+  try
+    {
+      instWrap.manifold_jacobian(refJacobian, input);
+    }
+  catch(std::runtime_error e)
+    {
+      std::cout << e.what() << std::endl;
+    }
```

Comment:
Well, maybe this part of the test needs to be rewritten, but it will probably stay more or less like this.
Why ? Because the call will fail in Sparse mode because the manifold of instWrap is a CartesianProduct, and the workaround for ApplyDiffRetractation is only implemented for RealSpaces manifolds. So the try catch is mandatory here, but as the call in Dense mode won't fail, we still need to do stuff to check it (absent stuff, by the way).

The perfect thing would possibly be to write something like :
if_Dense_mode (std::is_same<T,EigenMatrixDense> seems to do it right)
{
instWrap.manifold_jacobian(refJacobian, input);
BOOST_CHECK(refJacobian==expectedvalue) or (*output) << 
refJacobian
}
else
{
BOOST_CHECK_THROW(instWrap.manifold_jacobian(refJacobian, input));
}